cli/azd/pkg/tools/azcli/deployments.go

```diff
@@ -75,3 +78,127 @@ func (cli *azCli) createDeploymentsClient(
 
 	return client, nil
 }
+
+func (cli *azCli) DeployToSubscription(
+	ctx context.Context, subscriptionId, deploymentName, compiledBicep, parametersFile, location string) (
+	AzCliDeploymentResult, error) {
+	deploymentClient, err := cli.createDeploymentsClient(ctx, subscriptionId)
+	if err != nil {
+		return AzCliDeploymentResult{}, fmt.Errorf("creating deployments client: %w", err)
+	}
+
+	templateJsonAsMap, err := readFromString([]byte(compiledBicep))
```

Comment:
The docs for the API make me think this can just be the JSON itself, we don't need to move to `map[string]any`.  Did that not work?

Reply:
I tried the Json string directly, but the request failed.
It took me some time to find this example from the SDK itself where they parse the JSON data into a `map[string]any`
https://github.com/Azure-Samples/azure-sdk-for-go-samples/blob/main/sdk/resourcemanager/resource/deployment/main.go#L81

Reply:
the sdk calls `runtime.MarshalAsJSON()` and ended up making a double scaping to the the json string (because the string returned after compiling bicep is already scaped).